[CORE-11450] QoS controls: update docs for peakrate, minburst and packet burst

coutinhop opened the pull request:
Update the QoS controls docs for the newly added ingress and egress configurations: peakrate, minburst and packet burst.

Update minimum, maximum and default values of some QoS controls according to code changes.

<!--- PR title format: [GH#<gh-issue-id>][DOCS#<jira-issue-id>]: <short-description-of-the-pr> --->

Product Version(s):
<!--- Specify which versions of Calico, Calico Enterprise, and Calico Cloud your PR applies to. -->

Issue:
<!--- Add a link to the Jira ticket or GitHub issue, if applicable. --->

Link to docs preview:
<!--- Add direct link(s) to the exact page(s) with updated content from the preview build. --->

SME review:
- [ ] An SME has approved this change. 
<!--- SME approval is required to merge a PR except for changes that do not impact the meaning of the docs. --->

DOCS review:
- [ ] A member of the docs team has approved this change.
<!-- The Docs team must review and approve all changes. -->

Additional information:
<!--- Optional: Include additional context or expand the description here.--->

Merge checklist:
<!--- Mandatory for docs team members before merging code --->
- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed 


<!--- After you open your PR, ask for review from docs team:
  For community authors: tag @tigera/docs to ask for a review when your PR is ready --->

coutinhop committed:
[CORE-11450] QoS controls: update docs for peakrate, minburst and packet burst

Update the QoS controls docs for the newly added ingress and egress
configurations: peakrate, minburst and packet burst.

Update minimum, maximum and default values of some QoS controls
according to code changes.

ctauchen commented on calico-enterprise/networking/configuring/qos-controls.mdx:
```diff
@@ -62,55 +62,72 @@ spec:
+
+## Additional resources
+
+- [QoS controls on Openstack](../openstack/neutron-api.mdx#qos)
```
CE doesn't support OpenStack, so I'm not sure we want to have this section. In any case, it's causing a build error.

ctauchen commented on calico-enterprise/networking/configuring/qos-controls.mdx:
```diff
@@ -10,7 +10,7 @@ Configure QoS (Quality of Service) Controls to limit ingress and/or egress bandw
 ## Value
 
-With QoS controls, $[prodname] enforces network resource limits (bandwidth, packet rate, connections) for Kubernetes pods, ensuring fair resource allocation and preventing performance degradation for other workloads.
+With QoS controls, $[prodname] enforces network resource limits (bandwidth, packet rate, connections) for Kubernetes pods and OpenStack VMs, ensuring fair resource allocation and preventing performance degradation for other workloads.
```
Are we supporting OpenStack for CE? I'm not sure we want to add this. A reader would reasonably assume that there's support.

coutinhop replied:
Right, I was wrong to just copy the OSS page over to EE, I'll remove the OpenStack references. Thanks for catching it!

ctauchen commented on calico-enterprise/networking/configuring/qos-controls.mdx:
```diff
@@ -62,55 +62,72 @@ spec:
 ### Configure bandwidth limits
 
-Use the annotations below to configure bandwidth and burst limits for traffic going into or coming out of the pod. Limiting the bandwidth enforces that, on average, that will be the maximum bit rate used. Limiting the burst enforces that at any given instant, the maximum burst traffic will not go over the burst limit. Burst configuration is optional, if burst is not explicitly configured with an  ...
+Use the annotations below to configure bandwidth and burst limits for traffic going into or coming out of the pod. Limiting the bandwidth enforces that, on average, that will be the maximum bit rate used. The burst size is the amount of data that is available instantaneously for transmission. Burst configuration is optional, if burst is not explicitly configured with an annotation, a default valu ...
```
* s/peakrate/peak rate/g
* Suggestion: If you configure a peak rate limit, burst traffic will not go over that limit. Otherwise, there are no limits to burst size.

ctauchen commented on calico-enterprise/networking/configuring/qos-controls.mdx:
```diff
@@ -62,55 +62,72 @@ spec:
+| `qos.projectcalico.org/egressBandwidth`       | Specifies the egress bandwidth limit (in bits per second).<br/>Min value: `1k`<br/>Max value: `1P`                                                                                                                             |
+| `qos.projectcalico.org/ingressBurst`          | Specifies the ingress burst size (in bits).<br/>Optional, defaults to `4Gi`.<br/>Min value: `1k`<br/>Max value: `34359738360`                                                                                                    |
+| `qos.projectcalico.org/egressBurst`           | Specifies the egress burst size (in bits).<br/>Optional, defaults to `4Gi`.<br/>Min value: `1k`<br/>Max value: `34359738360`                                                                                                     |
+| `qos.projectcalico.org/ingressPeakrate`       | Specifies the ingress peakrate, i.e., the bandwidth limit during bursts (in bits per second).<br/>Optional, defaults to unlimited.<br/>Must be higher than the bandwidth limit<br/>Min value: `1.01k`<br/>Max value: `1.01P`   |
```
```suggestion
| `qos.projectcalico.org/ingressPeakrate`       | Specifies the ingress peak rate limit, i.e., the bandwidth limit during bursts (in bits per second).<br/>Optional, defaults to unlimited.<br/>Must be higher than the bandwidth limit.<br/>Min value: `1.01k`<br/>Max value: `1.01P`   |
```

ctauchen commented on calico-enterprise/networking/configuring/qos-controls.mdx:
```diff
@@ -62,55 +62,72 @@ spec:
+| `qos.projectcalico.org/ingressBurst`          | Specifies the ingress burst size (in bits).<br/>Optional, defaults to `4Gi`.<br/>Min value: `1k`<br/>Max value: `34359738360`                                                                                                    |
+| `qos.projectcalico.org/egressBurst`           | Specifies the egress burst size (in bits).<br/>Optional, defaults to `4Gi`.<br/>Min value: `1k`<br/>Max value: `34359738360`                                                                                                     |
+| `qos.projectcalico.org/ingressPeakrate`       | Specifies the ingress peakrate, i.e., the bandwidth limit during bursts (in bits per second).<br/>Optional, defaults to unlimited.<br/>Must be higher than the bandwidth limit<br/>Min value: `1.01k`<br/>Max value: `1.01P`   |
+| `qos.projectcalico.org/egressPeakrate`        | Specifies the egress peakrate, i.e., the bandwidth limit during bursts (in bits per second).<br/>Optional, defaults to unlimited.<br/>Must be higher than the bandwidth limit<br/>Min value: `1.01k`<br/>Max value: `1.01P`    |
```
see above pattern

ctauchen commented on calico-enterprise/networking/configuring/qos-controls.mdx:
```diff
@@ -62,55 +62,72 @@ spec:
 ### Configure packet rate limits
 
-Use the annotations below to configure packet rate limits for traffic going into or coming out of the pod. Limiting the packet rate enforces that, on average, that will be the maximum rate that packets will be sent and/or received by the pod.
+Use the annotations below to configure packet rate limits for traffic going into or coming out of the pod. Limiting the packet rate enforces that, on average, that will be the maximum rate that packets will be sent and/or received by the pod. The packet burst size is the number of packets that are available instantaneously for transmission. Packet burst configuration is optional, if packet burst  ...
```
```suggestion
Use the annotations below to configure packet rate limits for traffic going into or coming out of the pod. Limiting the packet rate enforces that, on average, that will be the maximum rate that packets will be sent and/or received by the pod. The packet burst size is the number of packets that are available instantaneously for transmission. Packet burst configuration is optional. If packet burst is not explicitly configured with an annotation, a default value of `5` packets is used.
```

ctauchen reviewed (commented):
A few comments and questions for you. Thanks!

coutinhop committed:
address review comments: peakrate -> peak rate, etc

coutinhop commented:
> A few comments and questions for you. Thanks!

Thanks for catching those, I did address them, please have another look when you can.

coutinhop committed:
Address test cycle comments:
- Remove repeated "all annotations" table from QoS docs
- Group annotations by "ingress" and "egress" as it makes more sense

Additionally, roll back changes in the v3.21-2 versioned docs as those
should be merged only on a v3.21.1 patch release.

ctauchen commented:
Thanks, @coutinhop LGTM. Is this going out with 3.30.2? If so, I'll merge to the publication branch. Otherwise can merge immediately.

coutinhop commented:
Thanks @ctauchen! Yes, it's going out with v3.30.2
